### PR TITLE
Array.prototype.find, findIndex, fill

### PR DESCRIPTION
**Pull request by damz**
Implements `Array.prototype.find()`, `Array.prototype.findIndex()` and `Array.prototype.fill()` from ES6.

### CODE REVIEW

File: builtin_array.go [hunk 1]
```diff
@@ -681,3 +681,52 @@ func builtinArray_reduceRight(call FunctionCall) Value {
 	}
 	panic(call.runtime.panicTypeError())
 }
+
+func builtinArray_find(call FunctionCall) Value {
+	thisObject := call.thisObject()
+	this := toValue_object(thisObject)
+	if iterator := call.Argument(0); iterator.isCallable() {
```
Comment (ralt) [inline builtin_array.go]: You don't need `ToLength()` there?

Comment (damz) [reply builtin_array.go]: Yes, of course. I added three tests for exotic objects too.

File: array_test.go [hunk 2]
```diff
@@ -689,6 +689,63 @@ func TestArray_reduceRight(t *testing.T) {
...
+		test(`[].find(function() { return true })`, Value{})
+
+		test(`[1,2,3].find(function() { return false })`, Value{})
+
+		test(`[1,2,3].find(function() { return true })`, 1)
+
+		test(`[1,2,3].find(function(_, index) { if (index === 1) return true })`, 2)
+
+		test(`Array.prototype.find.call({length: -1}, function(v) { return v === "bar"; })`, Value{})
+		test(`Array.prototype.find.call({0: "foo", 1: "bar", 2: "baz", length: 3}, function(v) { return v === "bar"; })`, "bar")
```
Comment (ralt) [inline array_test.go]: What does `Array.prototype.find.call({}, x => x)` return?

Comment (damz) [reply array_test.go]: It will return `undefined`. The equivalent `findIndex` would return `-1`. I added an explicit test.

**Pull request merged**